mono/utils/dlmalloc.c

```diff
@@ -1187,7 +1187,7 @@ int mspace_mallopt(int, int);
 #include <string.h>      /* for memset etc */
 #endif  /* LACKS_STRING_H */
 #if USE_BUILTIN_FFS
-#ifndef LACKS_STRINGS_H
+#ifndef LACKS_STRINGS_H && defined(HAVE_STRINGS_H)
```

Comment:
I don't think `#ifndef` composes that way, but is merely ignoring the rest of the line.
It should be `#if !defined (LACKS_STRINGS) && defined (HAVE_STRINGS_H)`.

Reply:
Or does not even compile.

Reply:
```
$ cat 1.c
#ifndef LACKS_STRINGS_H && defined(HAVE_STRINGS_H)
#endif
HB30-VM-Eklavya:mono4 jay$ gcc -c  1.c
1.c:1:25: warning: extra tokens at end of #ifndef directive [-Wextra-tokens]
#ifndef LACKS_STRINGS_H && defined(HAVE_STRINGS_H)
                        ^
                        //
1 warning generated.
```